src/main/java/com/demo/classroom/Controller/AuthController.java

```diff
@@ -52,4 +54,41 @@ public ResponseEntity<ApiResponse<?>> register(
             return ResponseEntity.badRequest().body(apiResponse);
         }
     }
+
+
+    @PostMapping(value = "/login", consumes = "application/json")
+    public ResponseEntity<ApiResponse<?>> login(@Valid @RequestBody LoginDTO request, BindingResult result) {
+
+        if (result.hasErrors()) {
+            Map<String, List<String>> errors = ErrorMessages.constructErrorMessages(result);
+            ApiResponse<Map<String, List<String>>> errorResponse = new ApiResponse<>(
+                false, 
+                Constants.VALIDATION_FAILED.getMessage(), 
+                errors
+            );
+            return ResponseEntity.badRequest().body(errorResponse);
+        }
+
+        ApiResponse<?> apiResponse = authService.loginUser(request);
+
+        if (apiResponse.isSuccess()) {
+            return ResponseEntity.status(HttpStatus.CREATED).body(apiResponse);
+        } else {
+            return ResponseEntity.badRequest().body(apiResponse);
+        }
+    
+    }
+
+    @PostMapping("/refresh-token")
+    public ResponseEntity<ApiResponse<?>> refreshToken(@RequestHeader("Refresh-Token") String refreshToken) {
```

Comment:
nstead of returning `ResponseEntity<ApiResponse<?>>`, consider using a more specific type, like `ResponseEntity<ApiResponse<Map<String, String>>>` in the `/refresh-token` method to make the response type explicit.

Reply:
Solved